### PR TITLE
[Can] test-backend

### PR DESCRIPTION
**Pull request by Soul-Clinic**
Most backend codes are on database. 
There are lots of front-end codes, BTW (Designed by myself :). 
I tried to use as less 3-party frameworks as possible (like Vue/Koa/Express...)

### CODE REVIEW

**Comment by xavierchow** [comment]
@Soul-Clinic 
> I tried to use as less 3-party frameworks as possible (like Vue/Koa/Express...)

What's the rationale for the design decision above?

**Comment by Soul-Clinic** [comment]
Experience from last years to this year.
I try to use: 
web-socket instead of http request, 
web-component instead of Vue, 
functions about data write in database(postgres) instead of in node/lisp... 

I feel it's cool and efficient, so I insist. 
Just for Fun.

File: node/app.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+const PORT = 1954
+const { SERVER } = process.env
+const $remote = SERVER === 'celwk'
+const domain = $remote ? 'https://wcraft.canskit.com' : 'https://canskit.wcraft.cn' // Antoher Lisp Version: https://wcraft.celwk.com
+HTTP.createServer(async (req, res) => {
+    const { headers, url, method } = req
+    console.log(headers)
+    // console.log({ headers, url, method })
+
+    res.writeHead(200, {'Content-Type': 'application/json'})
```
Comment (xavierchow) [inline node/app.js:12]: does it mean for any request the HTTP response code would be 200?

File: node/tool.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
+const { Pool } = require('pg');
+const crypto = require('crypto');
+String.prototype.underline = function () {
+    return this.replace(/-/g, '_')
+}
+global.sql = (literals, ...params) => {    // For the syntax highlight in VSCode plugin: SQL Lit
```
Comment (xavierchow) [inline node/tool.js:6]: any reason to use `global` here?

File: node/tool.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    const decipher = crypto.createDecipher('aes192', key);
+    var decrypted = decipher.update(encrypted, 'hex', 'utf8');
+    decrypted += decipher.final('utf8');
+    return decrypted;
+}
+const pool = new Pool({
+    database: 'wiredcraft'
+}) 
+
+const $query = async (...args) => {
```
Comment (xavierchow) [inline node/tool.js:39]: how to prevent the SQL injection?

File: node/tool.js (hunk 2 above)
Comment (Soul-Clinic) [reply node/tool.js:6]: sql`SELECT... ` is a syntax highlight sugar, so I make it global.

File: node/tool.js (hunk 3 above)
Comment (Soul-Clinic) [reply node/tool.js:39]: The arguments in `pg` will be encapsulated with $1, $2, $3...

File: node/app.js (hunk 1 above)
Comment (Soul-Clinic) [reply node/app.js:12]: For testing purpose, I haven't paid attention to authentication yet.(That's not for this project specially?)

**Comment by Soul-Clinic** [comment]
I've tried to make it clearer, most functions except `login` are implement on database, however.

**Comment by xavierchow** [comment]
Thanks for your PR @Soul-Clinic 

I'm gonna close it and we will get back to you soon.

**Pull request closed without merging**